<launch>
  <arg name="bag_name"/>
  <arg name="s" default="0" />
  <arg name="viz" default="false" />

  <node pkg="mimosa" type="mimosa_rosbag" name="mimosa_node" output="screen" clear_params="true"> # launch-prefix="gdb -ex run --args">
    <remap from="/mimosa_node/imu/manager/imu_in" to="/ouster/imu" />
    <remap from="/mimosa_node/lidar/manager/points_in" to="/ouster/points" />

    <param name="bag_name" value="$(arg bag_name)"/>
    <param name="s" value="$(arg s)"/>

    <rosparam file="$(find mimosa)/config/enwide/params.yaml"/>
    <rosparam file="$(find mimosa)/config/enwide/os_enwide.json" command="load" ns="lidar/sensor"/>
  </node>

  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find mimosa)/rviz/mimosa.rviz" if="$(arg viz)" />

</launch>
